modules/compute/gke-node-pool/main.tf

```diff
@@ -66,13 +66,25 @@ resource "google_container_node_pool" "node_pool" {
     max_unavailable = 1
   }
 
+  # NOTE: `placement_type` and `placement_policy` together can support compact placement of nodes in a node pool.
+  # So, one might wonder if we can remove this block. But to support backward compatibility this needs to be kept.
+  # By backward compatibility we mean that there maybe blueprints that might be using `compact_placement`.
+  # Those blueprints will stop working if we remove this block.
   dynamic "placement_policy" {
     for_each = var.compact_placement ? [1] : []
     content {
       type = "COMPACT"
     }
   }
 
+  dynamic "placement_policy" {
+    for_each = (!var.compact_placement && try(contains(["COMPACT"], var.placement_type), false)) ? [1] : []
```

Comment:
nit: If we are checking that COMPACT is the only valid placement_type we are accepting in the variables, then we don't need to verify that again.

Reply:
Ok. Let me check. I agree this can be removed as long as it's redundant.